### DISCUSSION TITLE
Retrieving ALL Messages

### DISCUSSION THREAD

**Discussion by BasselAshi** [Auties00/WhatsappWeb4j#404] [General]
How do I retrieve all in a chat given that it has probably over 100k messages?
* I attempted using `WebHistoryLength` with no changes appearing
* I could not find any paging implementation for `messages()` or a different way of retrieving older messages than the ones provided

**Comment by Auties00** [comment]
> How do I retrieve all in a chat given that it has probably over 100k messages?
> * I attempted using `WebHistoryLength` with no changes appearing
> * I could not find any paging implementation for `messages()` or a different way of retrieving older messages than the ones provided

Good question, I think it can be done using the history sync config field. I'll push it when cobalt 0.1 is ready.  Still I can't guarantee it, it's up to WhatsApp's servers to send the messages

**Comment by Auties00** [comment]
I can confirm that it works, I'll push out this feature soon.
It could still be limited tho.
It will be available for everyone in Cobalt 0.1

**Reply by Felic1ty** [reply]
Do you have any relevant examples?
